stages/org.osbuild.kickstart

```diff
@@ -197,6 +236,29 @@ def make_users(users: Dict) -> List[str]:
     return res
 
 
+def make_clearpart(options: Dict) -> str:
+    clearpart = options.get("clearpart")
+    if clearpart is None:
+        return ""
+    cmd = "clearpart"
+    al = clearpart.get("all", False)
+    if al:
+        cmd += " --all"
+    drives = clearpart.get("drives", [])
+    if drives:
+        cmd += f" --drives={','.join(drives)}"
```

Comment:
These each have the regular expression `^[A-Za-z0-9/:\\_|*-]+$` for valid contents; I'm not 100% on kickstart but if this is passed through a shell at some point we might want to `shlex.quote` each value.

@bcl do these values need to be escaped in kickstart files in some way (either for shell or in another way?).

Reply:
> @bcl do these values need to be escaped in kickstart files in some way (either for shell or in another way?).

No, they should be left as-is. The specifiers are parsed by kickstart into a python list which is passed to anaconda for processing by this code https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/core/storage.py#L139 so if you add extra quoting things won't match.

Reply:
Thank you then this is resolved :)